test/framework/go.mod

```diff
@@ -0,0 +1,16 @@
+module sigs.k8s.io/cluster-api/test/framework
+
+go 1.13
+
+require (
+	github.com/onsi/ginkgo v1.10.3
```

Comment:
What's the reason for as a separate module here?

Reply:
I made this a separate module so i didn't have to worry about polluting cluster-api's go.mod file with things like `sigs.k8s.io/kind`, as that is not a dependency of cluster-api but is a dependency of this framework.

Reply:
Makes sense, I don't see `sigs.k8s.io/kind` in this file though?

Reply:
oh, you're right, i only needed it in the docker provider. 

Yeah, I could see it going either way then.

Is there some benefit to keeping this in the cluster-api module?

Reply:
If the dependencies are identical why manage both? Most likely this module will be forgotten when the root module is being updated so there will be drift between the two often.

Reply:
We already ran into an issue with cluster-api-upgrade-tool's e2e test where the test code was pulling in a dependency that used a different version of client-go, and we decided to create a separate e2e test go module so the main module could be kept clean and not require any replace directives. I'm fine either way on this (1 or 2 modules) (although personally I'm in favor of 2); however, the minute that we find we're having transitive dependency issues, potentially requiring use to use a replace directive in the main module, that would be the time to split into 1 main and 1 test module.

Reply:
We will likely be moving away from exec entirely which will pull in a good number of dependencies. I think we should keep this a separate module as so not to pollute the main go.mod